microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/commons/CodeActionData.java

```diff
@@ -0,0 +1,103 @@
+/*******************************************************************************
+* Copyright (c) 2023 Red Hat Inc. and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4mp.commons;
+
+import java.util.Objects;
+
+/**
+ * Represents the data that all code actions have.
+ */
+public class CodeActionData {
+
+	public static enum CodeActionId {
+		UnknownPropertySimilarTextSuggestion, //
```

Comment:
I suggest that CodeActionId stores the title too, il xill simplify the code.

IN other words i suggest  that you see https://github.com/redhat-developer/quarkus-ls/blob/4dc6eb085250d33baf3f0707d1027f3655cf473f/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/diagnostics/QuteErrorCode.java

Reply:
I started doing this, but not every CodeAction has an associated template. For instance, there are many different code actions that insert an annotation (insert health annotation, insert inject annotation, ...) but there is only one template used for all these different code actions, and it's used in an abstract class. Since there isn't a 1 to 1 mapping of code action type to title template, I'm not sure this is worthwhile.